src/main/java/com/softserve/demo/util/validators/OrderDatesChecker.java

```diff
@@ -0,0 +1,22 @@
+package com.softserve.demo.util.validators;
+
+import com.softserve.demo.dto.OrderDTO;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+@Slf4j
+public class OrderDatesChecker implements ConstraintValidator<OrderDateCheck, OrderDTO> {
```

Comment:
Offer and order date checkers are the same, so, may be, we can extract common interface for this 2 fields and have only on Checker, that accepts interface? Then we can reuse this validation in all classes, that have start/end date.
See
https://stackoverflow.com/questions/33873163/java-303-349-start-date-before-end-date-validation